### PR TITLE
Dreamview:default cycle routing

### PR DESCRIPTION
**Pull request by lj0919**
add cycle routing at DV module
screenshot
①add task_manager module
image
②add choose default routing point function at route editing bar
image
③default routing name input to input user-defined default routing name and save it in a txt file
image

④load default routing from the txt file and show it on the tool panel like poi
image
⑤input cycle number(default 1) and send cycle routing request to task manager 
image

### CODE REVIEW

File: modules/dreamview/frontend/src/renderer/routing_editor.js [hunk 1]
```diff
@@ -47,6 +52,25 @@ export default class RoutingEditor {
     WS.checkRoutingPoint(point);
   }
 
+  drawDefaultRoutingPath(point, coordinates, scene) {
```
Comment (lj0919) [inline modules/dreamview/frontend/src/renderer/routing_editor.js:55]: @vlin17

File: modules/dreamview/backend/simulation_world/BUILD [hunk 2]
```diff
@@ -68,6 +68,7 @@ cc_library(
         "//modules/dreamview/backend/perception_camera_updater",
         "//modules/dreamview/backend/sim_control",
         "//modules/routing/proto:poi_cc_proto",
+         "//modules/routing/proto:default_routing_cc_proto",
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/BUILD:71]: notice the format and the order

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc [hunk 3]
```diff
@@ -237,6 +269,40 @@ void SimulationWorldUpdater::RegisterMessageHandlers() {
...
+            Json drouting;
+            drouting["name"] = defaultrouting.name();
+            Json point_list;
+            for (const auto &point : defaultrouting.point()) {
+              Json point_json;
+              point_json["x"] = point.x();
+              point_json["y"] = point.y();
+              point_json["z"] = point.z();
+              point_list.push_back(point_json);
+
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.cc:290]: remove the blank line

File: modules/dreamview/backend/simulation_world/simulation_world_updater.h [hunk 4]
```diff
@@ -22,6 +22,7 @@
 
 #include <memory>
 #include <string>
+#include<algorithm>
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.h:25]: `#include <algorithm>`, and notice the order

**Comment by lj0919** [comment]
@knzeus

File: modules/dreamview/frontend/src/renderer/index.js [hunk 5]
```diff
@@ -534,4 +550,4 @@ class Renderer {
 
 const RENDERER = new Renderer();
 
-export default RENDERER;
+export default RENDERER;
```
Comment (changsh726) [inline modules/dreamview/frontend/src/renderer/index.js:553]: leave an empty line here

File: modules/dreamview/backend/simulation_world/simulation_world_updater.cc [hunk 6]
```diff
@@ -160,6 +165,35 @@ void SimulationWorldUpdater::RegisterMessageHandlers() {
...
+          return;
+        }
+        bool succeed = ConstructRoutingRequest(json, routing_request);
+        if (succeed) {
+          cycle_routing_task->set_cycle_num(
+              static_cast<int>(json["cycleNumber"]));
+          task->set_task_name("cycle_routing_task");
+          task->set_task_type(apollo::task_manager::TaskType::CYCLE_ROUTING);
+          sim_world_service_.PublishTask(task);
+          AINFO << task->DebugString();
```
Comment (changsh726) [inline modules/dreamview/backend/simulation_world/simulation_world_updater.cc:187]: ADEBUG?

File: modules/dreamview/backend/simulation_world/simulation_world_updater.h [hunk 7]
```diff
@@ -20,6 +20,7 @@
 
 #pragma once
 
+#include <algorithm>
```
[the remark this answers is not in the document]
Comment (lj0919) [reply modules/dreamview/backend/simulation_world/simulation_world_updater.h:23]: will be used in simulation_world_updater.cc

[the remark this answers is not in the document]
Comment (changsh726) [reply modules/dreamview/backend/simulation_world/simulation_world_updater.h:23]: Haven't find a place to use it, and target `simulation_world_updater` can be built without including this.

**Comment by MorphZhou** [comment]
Looks good to me. Remember to update the screenshot.

**Review by changsh726 — APPROVED** [review]

**Pull request merged**